enterprise/server/githubapp/githubapp.go

```diff
@@ -204,7 +244,122 @@ func (a *GitHubApp) handleInstallationEvent(ctx context.Context, eventType strin
 	return nil
 }
 
-func (a *GitHubApp) handleWorkflowEvent(ctx context.Context, eventType string, event any) error {
+func (a *GitHubApp) handleWorkflowJobEvent(ctx context.Context, eventType string, event *github.WorkflowJobEvent) error {
```

Comment:
I would be more explicit about calling these githubWorkflow events.

For better or worse workflow jobs already means something to us all, so it's kind of confusing to change the meaning of that name now.

Reply:
Renamed `startWorkflowJob` to `startGitHubActionsRunnerTask`.

I would like to keep this func `handleWorkflowJob` the same though - in `handleWebhookEvent` above, the function that handles `*github.FooEvent` is called `handleFooEvent` which I think is a nice convention. To make this more obvious, the latest commit also gets rid of `handleBuildBuddyWorkflowEvent` which was being used as a catch-all for the remaining event types, which seemed sort of unclear. Replaced that with explicit handlers for Push, PullRequest, and PullRequestReview events, and those handlers call `maybeTriggerBuildBuddyWorkflow` now